src/lava/lib/optimization/solvers/generic/solver.py

```diff
@@ -156,9 +190,26 @@ def solve(self,
         )
         if timeout == -1:
             self.solver_process.wait()
-        solution = self.solver_process.variable_assignment.aliased_var.get()
+        self._update_report(target_cost=target_cost)
         self.solver_process.stop()
-        return solution
+        return self._report["best_state"]
+
+    def _update_report(self,
+                       target_cost=None):
+        self._report["target_cost"] = target_cost
+        best_state = self.solver_process.variable_assignment.aliased_var.get()
+        self._report["best_state"] = best_state
+        raw_cost = self.solver_process.optimality.aliased_var.get()
+        cost = (raw_cost.astype(np.int32) << 8) >> 8
+        self._report["cost"] = cost
+        self._report["solved"] = cost == target_cost

```

Comment:
Is it solved if cost == target_cost or <=?

Reply:
I would say cost == target_cost, but @GaboFGuerra should confirm.

Reply:
should be <= as it will imply an even better solution. Otherwise you would not stop unless you are at exactly the target cost.

Reply:
Makes sense